add note for automatic issue management configuration (#3429)
getsentry/sentry-docs: src/docs/product/integrations/jira/index.mdx

#### Automatically

<Note>

> This configuration method does not apply to Jira Server. For Jira Server, go to [the documentation on manual configuration](#manually).

Reviewer: ```suggestion
This configuration method does not apply to Jira Server. Jira Server [requires manual configuration](#manually).
```
